doc/format_specifications.rdoc

```diff
@@ -163,34 +180,126 @@ Format +argument+ as a binary integer:
 
   sprintf('%b', 1)  # => "1"
   sprintf('%b', 4)  # => "100"
+
   # Prefix '..' for negative value.
   sprintf('%b', -4) # => "..100"
 
-Type-specific modifier:
+  # Alternate format.
+  sprintf('%#b', 4)  # => "0b100"
+  sprintf('%#B', 4)  # => "0B100"
 
-- '#' flag' (use alternate format):
+==== Specifier +c+
 
-    sprintf('%#b', 1)  # => "0b1"
-    sprintf('%#B', 1)  # => "0B1"
-    sprintf('%#b', 4)  # => "0b100"
+Format +argument+ as a single character:
 
-==== Specifier +c+
+  sprintf('%c', 'A') # => "A"
+  sprintf('%c', 65)  # => "A"
+
+Flags do not apply.
 
 ==== Specifier +d+
 
+Format +argument+ as a decimal integer:
+
+  sprintf('%d', 100)  # => "100"
+  sprintf('%d', -100) # => "-100"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +e+ and +E+
 
+The two specifiers +e+ and +E+ behave identically
+except when flag <tt>'#'</tt>+ is used.
+
+Format +argument+ in
+{scientific notation}[https://en.wikipedia.org/wiki/Scientific_notation]:
+
+  sprintf('%e', 3.14159)  # => "3.141590e+00"
+  sprintf('%e', -3.14159)  # => "-3.141590e+00"
+  sprintf('%#e', 3.14159)  # => "3.141590e+00"
+  sprintf('%#E', 3.14159)  # => "3.141590E+00"
+
 ==== Specifier +f+
 
+Format +argument+ as a floating-point number:
+
+  sprintf('%f', 3.14159)  # => "3.141590"
+  sprintf('%f', -3.14159) # => "-3.141590"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +g+ and +G+
 
+Format +argument+ in a "general" format that may be convenient
+for human readers:
+
+  sprintf('%g', 100)  # => "100"
+  sprintf('%g', 100.0)  # => "100"
+  sprintf('%g', 3.14159)  # => "3.14159"
+  sprintf('%g', 100000000000)  # => "1e+11"
+  sprintf('%g', 0.000000000001)  # => "1e-12"
+
+  # Capital 'G' means use capital 'E'.
+  sprintf('%G', 100000000000)  # => "1E+11"
+  sprintf('%G', 0.000000000001)  # => "1E-12"
+
+  # Alternate format.
+  sprintf('%#g', 100000000000)  # => "1.00000e+11"
+  sprintf('%#g', 0.000000000001)  # => "1.00000e-12"
+  sprintf('%#G', 100000000000)  # => "1.00000E+11"
+  sprintf('%#G', 0.000000000001)  # => "1.00000E-12"
+
 ==== Specifier +o+
 
+Format +argument+ as an octal integer:
+
+  sprintf('%o', 16)   # => "20"
+
+  # Prefix '..' for negative value.
+  sprintf('%o', -16)  # => "..760"
+
+  # Prefix zero for alternate format if positive.
+  sprintf('%#o', 16)  # => "020"
+  sprintf('%#o', -16) # => "..760"
+
 ==== Specifier +p+
 
+Format +argument+ as a string via <tt>argument.inspect</tt>:
+
+  t = Time.now
+  sprintf('%p', t)   # => "2022-05-01 13:42:07.1645683 -0500"
+
+  # Alternate format.
+  sprintf('%#p', t)  # => "2022-05-02 14:19:45.71561 -0500"
+
 ==== Specifier +s+
 
+Format +argument+ as a string via <tt>argument.to_s</tt>:
+
+  t =Time.now
+  sprintf('%s', t) # => "2022-05-01 13:42:07 -0500"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +x+ and +X+
 
+Format +argument+ as a hexadecimal number:
```

Comment:
```suggestion
Format +argument+ as a hexadecimal integer.
If +argument+ is negative, it will be formatted as a two's complement
prefixed with +..f+:
```